sdcflows/utils/tools.py

```diff
@@ -23,6 +23,71 @@
 """Image processing tools."""
 
 
+def deoblique_and_zooms(in_reference, oblique, factor=4, padding=1):
+    """
+    Generate a sampling reference aligned with in_reference fully covering oblique.
+
+    Parameters
+    ----------
+    in_reference : :obj:`~nibabel.spatialimages.SpatialImage`
+        The sampling reference.
+    oblique : :obj:`~nibabel.spatialimages.SpatialImage`
+        The rotated coordinate system whose extent should be fully covered by the
+        generated reference.
+    factor : :obj:`int`
+        A factor to increase the resolution of the generated reference.
+    padding : :obj:`int`
+        Number of additional voxels around the most extreme positions of the projection of
+        oblique on to the reference.
+
+    """
+    from itertools import product
+    import numpy as np
+    from nibabel.affines import apply_affine, rescale_affine
+
+    # Reference space metadata
+    hdr = in_reference.header.copy()
+    affine = in_reference.affine
+    ref_shape = np.array(in_reference.shape[:3])
+    ref_zooms = np.array(hdr.get_zooms()[:3])
+    _, scode = in_reference.get_sform(coded=True)
+    _, qcode = in_reference.get_qform(coded=True)
+
+    # Calculate the 8 most extreme coordinates of oblique in in_reference space
+    extent = apply_affine(
+        oblique.affine,
+        np.array(list(product((0, 1), repeat=3))) * (ref_shape - 1),
+    )
+    extent_ijk = apply_affine(np.linalg.inv(affine), extent)
+
+    if isinstance(padding, int):
+        padding = tuple([padding] * 3)
+
+    underflow = min(extent_ijk.min(0) - padding, 0)
+    overflow = max(extent_ijk.max(0) + padding + 1 - ref_shape, 0)
```

Comment:
This doesn't make sense to me. I think you probably want to be using `np.min`/`np.max` or `np.minimum`/`np.maximum`.